enhancements/workload-partitioning/rationalizing-configuration.md

```diff
@@ -0,0 +1,213 @@
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The initial iteration of workload partitioning focused on a short path
+to a minimum viable implementation. This enhancement describes the
+loose ends for preparing the feature for GA at a high level, and
+explains the set of other design documents that need to be written
+separately during the next iteration.
+
+## Motivation
+
+The management workload partitioning feature introduced in OCP 4.8
+enables pod workloads to be pinned to the kubelet reserved cpuset. The
+feature is enabled at install time and requires the reserved set of
+CPUs as input. That configuration is currently passed in via a
+manually created manifest containing a machine config resource. The
+set of CPUs used for management workload partitioning has to align
+with the reserved set configured by the performance-addon-operator
+(PAO) during day 2. For OCP 4.8, the onus is on the user to ensure
+these align, there is no interlock.
+
+The current implementation is prone to user error, because it has two
+entities configuring and managing the same data. This design document
+describes phase 2 of the implementation plan for workload partitioning
+to improve the user experience when configuring the feature.
+
+### Goals
+
+1. Describe the remaining work at a high level for architectural
+   review and planning.
+2. Document the division of responsibility between workload management
+   in components in the OpenShift release payload and PAO.
+3. List the other enhancements that need to be written, and their
+   scopes.
+
+### Non-Goals
+
+1. Resolve all of the details for each of the next steps. We're not
+   going to talk a lot about names or implementation details here.
+
+## Proposal
+
+### API
+
+The initial implementation is limited to single-node deployments. To
+support multi-node clusters, we need to solve a couple of concurrency
+problems. The first is providing a way for the admission hook
+responsible for mutating pod definitions to reliably know when the
+feature should be enabled. The admission hook cannot store state on
+its own, so we need a new API to indicate when the feature is
+enabled.
+
+We eventually want to extend the partitioning feature to support
+non-management workloads to include workload types defined by cluster
+admins. For now, however, we want to keep the interface as simple as
+possible.  Therefore, the new API will be read-only, and only used so
+that the admission hook (and other consumers) can tell when workload
+partitioning is active.
+
+The new API will need a controller to manage it. We propose to add a
```

Comment:
I'm not really clear on why this API isn't an API where the user expresses his desire and the node changes flow from this desire.

Reply:
We expect to extend the PerformanceProfile CRD to be used to turn this on. We don't want the admission hook in the core to depend on that CRD, since it is part of the performance-addon-operator and may not be installed in all clusters.